zeppelin-jupyter/src/main/java/org/apache/zeppelin/jupyter/JupyterUtil.java

```diff
@@ -167,6 +164,100 @@ private Gson getGson(GsonBuilder gsonBuilder) {
+    return new Gson().toJson(note);
+  }
+
+  public String getNbformat(String note) {
+    Note noteFormat = getGson(new GsonBuilder()).fromJson(note, Note.class);
+
+    JsonObject nbformat = new JsonObject();
+    JsonArray cells = new JsonArray();
+
+    RegularExpression MD = new RegularExpression("%md\\s");
+    RegularExpression SQL = new RegularExpression("%sql\\s");
+    RegularExpression UNKNOWN_MAGIC = new RegularExpression("%\\w+\\s");
+    RegularExpression HTML = new RegularExpression("%html\\s");
+    RegularExpression SPARK = new RegularExpression("%spark\\s");
+
+    int index = 0;
+    for (Paragraph paragraph : noteFormat.getParagraphs()) {
+      String code = StringUtils.stripStart(paragraph.getText(), " ");
+      JsonObject codeJson = new JsonObject();
+
+      if (code == null || code.trim().isEmpty())
+        continue;
+
+      System.out.println(code);
+
+      if (MD.matches(code)) {
+        codeJson.addProperty("cell_type", "markdown");
+        codeJson.add("metadata", new JsonObject());
+        codeJson.addProperty("source",
+                StringUtils.stripStart(StringUtils.stripStart(code, "%md"),
+                        "\n"));  // remove '%md'
+      } else if (SQL.matches(code) || HTML.matches(code)) {
+        codeJson.addProperty("cell_type", "code");
+        codeJson.addProperty("execution_count", index);
+        codeJson.add("metadata", new JsonObject());
+        codeJson.add("outputs", new JsonArray());
+        codeJson.addProperty("source", "%" + code);  // add % to convert to cell magic
+      } else if (SPARK.matches(code)) {
+        codeJson.addProperty("cell_type", "code");
+        codeJson.addProperty("execution_count", index);
+        JsonObject metadataJson = new JsonObject();
+        metadataJson.addProperty("autoscroll", "auto");
+        codeJson.add("metadata", metadataJson);
+        codeJson.add("outputs", new JsonArray());
+        codeJson.addProperty("source", code);
+      } else if (UNKNOWN_MAGIC.matches(code)) {
+        // use raw cells for unknown magic
+        codeJson.addProperty("cell_type", "raw");
+        JsonObject metadataJson = new JsonObject();
+        metadataJson.addProperty("format", "text/plain");
+        codeJson.add("metadata", metadataJson);
+        codeJson.addProperty("source", code);
+      } else {
+        codeJson.addProperty("cell_type", "code");
+        codeJson.addProperty("execution_count", index);
+        JsonObject metadataJson = new JsonObject();
+        metadataJson.addProperty("autoscroll", "auto");
+        codeJson.add("metadata", metadataJson);
+        codeJson.add("outputs", new JsonArray());
+        codeJson.addProperty("source", code);
+      }
+
+      cells.add(codeJson);
+
+      index++;
+    }
+
+    JsonObject metadataJson = new JsonObject();
+
+    JsonObject kernelspecJson = new JsonObject();
+    kernelspecJson.addProperty("display_name", "Spark 2.0.0 - Scala 2.11");
```

Comment:
how is this going to be used? I don't think we should hardcode the Spark version number and Scala version

Reply:
Yeah I commented them as they weren't going to be used: https://github.com/apache/zeppelin/pull/2649/commits/deadcc77475c15b73470a87c9c07b4dd68b9236b